src/app.rs

```diff
@@ -111,13 +111,8 @@ impl<'a> App<'a> {
     // if playing and
     pub fn song_progress(&mut self) -> u16 {
         let progress = || {
-            let percentage =
-                (self.music_handle.time_played() * 100) / self.music_handle.song_length();
-            if percentage >= 100 {
-                100
-            } else {
-                percentage
-            }
+            let percentage: f32 = (self.music_handle.time_played() as f32 / self.music_handle.song_length() as f32) * 100.0;
+            min(percentage.round() as u16, 100)
```

Comment:
I did some testing with `u16::try_from(percentage).unwrap_or(100)` and other forms of converting numbers from/to different byte sizes and float/unsigned integers, and, even though `as X` seems to be discouraged, this was the only way I could get it right and in the most succinct way. 

Doing the division first guarantees that result will never over or underflow (as long as song_length isn't zero or some weird number I guess), and `min` is more succinct that `if x > 100 then 100 else x` IMO but that part's just a refactor and personal preference.

I don't know if the parenthesis are needed. I haven't looked into the default order of operations in Rust, but I think they communicate intention, so I'd leave them there — but maybe that's just me being a newbie to Rust.